### PR TITLE
Refactor licence summary page

### PR DESCRIPTION
**Pull request by jaucourt**
https://eaflood.atlassian.net/browse/IWTF-3099

Refactor licence summary page, bringing licence summary rows out of the nunjucks template and into the getData function so that they can be tested

### CODE REVIEW

File: packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js [hunk 1]
```diff
@@ -50,288 +39,250 @@ jest.mock('../../../../processors/mapping-constants.js', () => ({
...
-          birthDate: '1996-01-01'
-        },
-        isRenewal: true
-      }
+describe('licence-summary > route', () => {
+  beforeEach(jest.clearAllMocks)
 
-      mockTransactionCacheGet.mockImplementationOnce(() => permission)
+  describe('sets from summary on status cache', () => {
+    it('adds LICENCE_SUMMARY_SEEN to status.fromSummary for a renewal', async () => {
```
Comment (ScottDormand96) [inline packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js:168]: in this describe could we call statusCacheSet = jest.fn() once and use for all tests within?

File: packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js [hunk 2]
```diff
@@ -50,288 +39,250 @@ jest.mock('../../../../processors/mapping-constants.js', () => ({
...
-
-  describe('getFromSummary', () => {
-    it('should return licence-summary, if it is a renewal', async () => {
-      const result = await getFromSummary(undefined, true)
-      expect(result).toBe(LICENCE_SUMMARY_SEEN)
+const getMockRequest = ({ currentPermission = getMockPermission(), statusCache = {}, statusCacheSet = () => {} } = {}) => ({
+  cache: () => ({
+    helpers: {
+      status: {
+        getCurrentPermission: () => statusCache,
```
Comment (ScottDormand96) [inline packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js:48]: should be async or not?

File: packages/gafl-webapp-service/src/pages/summary/licence-summary/route.js [hunk 3]
```diff
@@ -45,6 +134,33 @@ export const checkNavigation = permission => {
...
+  rows.push(rg.generateStandardRow('licence_summary_name', `${licensee.firstName} ${licensee.lastName}`, NAME.uri, 'change-name'))
+  if (!permission.isRenewal) {
+    const birthDateStr = moment(permission.licensee.birthDate).locale(request.locale).format('Do MMMM YYYY')
+    rows.push(rg.generateStandardRow('licence_summary_dob', birthDateStr, DATE_OF_BIRTH.uri, 'change-birth-date'))
+  }
+  rows.push(
+    rg.generateStandardRow(
+      'licence_summary_type',
+      licenceTypeDisplay(permission, request.i18n.getCatalog()),
+      LICENCE_TYPE.uri,
```
Comment (ScottDormand96) [inline packages/gafl-webapp-service/src/pages/summary/licence-summary/route.js:151]: should we have language code added on here? same for when calling other EXAMPLE.uri

**Review by ScottDormand96 — CHANGES_REQUESTED** [review]

File: packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js (hunk 1 above)
Comment (jaucourt) [reply packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js:168]: No - as far as possible, you shouldn't share mocks between tests

File: packages/gafl-webapp-service/src/pages/summary/licence-summary/route.js (hunk 3 above)
Comment (jaucourt) [reply packages/gafl-webapp-service/src/pages/summary/licence-summary/route.js:151]: No, I add the language code on line 33, saves repeating it through the code then

File: packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js (hunk 2 above)
Comment (jaucourt) [reply packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js]: Yes, good spot (it doesn't seem to matter with tests, but it makes the intent clearer if we match the mock to the actual behaviour

**Review by ScottDormand96 — APPROVED** [review]

**Pull request merged**